File: android/superwardrobe/app/src/main/java/com/superwardrobe/data/model/OutfitDiary.kt
package com.superwardrobe.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OutfitDiary(
    val id: String = "",
    @SerialName("user_id") val userId: String = "",
    @SerialName("outfit_id") val outfitId: String? = null,
    val date: String = "",
    val mood: String? = null,
    val note: String? = null,
    @SerialName("photo_url") val photoUrl: String? = null,
    val weather: String? = null,
    val temperature: Double? = null,
    @SerialName("created_at") val createdAt: String = ""
)

File: ios/SuperWardrobe/SuperWardrobe/SuperWardrobe/Models/OutfitDiary.swift
import Foundation

struct OutfitDiary: Codable, Identifiable {
    let id: UUID
    var userId: UUID
    var date: Date
    var outfitId: UUID?
    var photoUrl: String?
    var note: String?
    var weatherData: String?
    var mood: String?
    var sharedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id, date, note, mood
        case userId = "user_id"
        case outfitId = "outfit_id"
        case photoUrl = "photo_url"
        case weatherData = "weather_data"
        case sharedAt = "shared_at"
    }
}
